WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java

```diff
@@ -172,11 +172,20 @@ public static ReaderPost fromJson(JSONObject json) {
         // xpost info
         assignXpostIdsFromJson(post, json.optJSONArray("metadata"));
 
-        // if the post doesn't have a featured image but it contains an IMG tag, check whether
-        // we can find a suitable image from the content
-        if (!post.hasFeaturedImage() && post.hasImages()) {
-            post.mFeaturedImage = new ReaderImageScanner(post.mText, post.isPrivate)
-                    .getLargestImage(ReaderConstants.MIN_FEATURED_IMAGE_WIDTH);
+        if (!post.hasFeaturedImage()) {
+            // if there's no featured image, check if featured media has been set to an image
+            if (json.has("featured_media")) {
+                JSONObject jsonMedia = json.optJSONObject("featured_media");
+                String type = JSONUtils.getString(jsonMedia, "type");
+                if (type.equals("image")) {
```

Comment:
If `type` is not an image then the image scan won't run and no featured image will be assigned. How about this instead:

```
// if there's no featured image, check if featured media has been set to an image
if (!post.hasFeaturedImage() && json.has("featured_media")) {
    JSONObject jsonMedia = json.optJSONObject("featured_media");
    String type = JSONUtils.getString(jsonMedia, "type");
    if (type.equals("image")) {
        post.mFeaturedImage = JSONUtils.getString(jsonMedia, "uri");
    }
}
// if the post doesn't have a featured media but it contains an IMG tag, check whether
// we can find a suitable image from the content
if (!post.hasFeaturedImage() && post.hasImages()) {
        post.mFeaturedImage = new ReaderImageScanner(post.mText, post.isPrivate)
                .getLargestImage(ReaderConstants.MIN_FEATURED_IMAGE_WIDTH);
}
```

Reply:
Makes sense, thanks.